lib/tests.sh

```diff
@@ -452,74 +452,86 @@ TEST_BACKUP_RESTORE () {
     # Check if an install have previously work
     at_least_one_install_succeeded || return 1
 
-    local check_path="$(default_install_path)"
-
-    # Install the application in a LXC container
-    _LOAD_SNAPSHOT_OR_INSTALL_APP "$check_path"
+    local check_paths=()
 
-    local ret=$?
+    if this_is_a_web_app; then
+        there_is_a_root_install_test && check_paths+=("$(root_path)")
+        there_is_a_subdir_install_test && check_paths+=("$(subdir_path)")
+    else
+        check_paths+=("")
+    fi
 
     local main_result=0
 
-    # Remove the previous residual backups
-    RUN_INSIDE_LXC rm -rf /home/yunohost.backup/archives
+    for check_path in "${check_paths[@]}"
+    do
+        # Install the application in a LXC container
+        _LOAD_SNAPSHOT_OR_INSTALL_APP "$check_path"
 
-    # BACKUP
-    # Made a backup if the installation succeed
-    if [ $ret -ne 0 ]
-    then
-        log_error "Installation failed..."
-    else
-        log_small_title "Backup of the application..."
+        local ret=$?
 
-        # Made a backup of the application
-        _RUN_YUNOHOST_CMD "backup create -n Backup_test --apps $app_id"
-        ret=$?
-    fi
+        # Remove the previous residual backups
+        RUN_INSIDE_LXC rm -rf /home/yunohost.backup/archives
 
-    [ $ret -eq 0 ] || main_result=1
+        # BACKUP
+        # Made a backup if the installation succeed
+        if [ $ret -ne 0 ]
+        then
+            log_error "Installation failed..."
+            main_result=1
+            continue
+        else
+            log_small_title "Backup of the application..."
 
-    # Grab the backup archive into the LXC container, and keep a copy
-    lxc file pull -r $LXC_NAME/home/yunohost.backup/archives $TEST_CONTEXT/ynh_backups
+            # Made a backup of the application
+            _RUN_YUNOHOST_CMD "backup create -n Backup_test --apps $app_id"
+            ret=$?
+        fi
+    
+        [ $ret -eq 0 ] || main_result=1
```

Comment:
if backup failed, do we really want to continue?
```suggestion
        [ $ret -eq 0 ] || { main_result=1; continue; }
```